File: picture-server/src/main/java/com/applory/pictureserver/shared/SecurityUtils.java
package com.applory.pictureserver.shared;

import org.springframework.security.core.context.SecurityContextHolder;

import java.util.Objects;

public class SecurityUtils {
    public static String getPrincipal() {
        if (Objects.nonNull(SecurityContextHolder.getContext().getAuthentication())) {
            return (String) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
        }

        return null;
    }
}

File: picture-server-kt/src/main/kotlin/com/applory/pictureserverkt/shared/SecurityUtils.kt
package com.applory.pictureserverkt.shared

import org.springframework.security.core.context.SecurityContextHolder

class SecurityUtils {
    companion object {
        @JvmStatic fun getPrincipal(): String {
            return SecurityContextHolder.getContext().authentication.principal.toString()
        }
    }
}
